File: data-ts/src/apy_bsol.ts
const BASE_URL = "https://stake.solblaze.org";

interface SolBlazeApyResponse {
  success: boolean;
  apy: number;  
}


export async function getSolBlazeApy(): Promise<SolBlazeApyResponse> {
  const url = `${BASE_URL}/api/v1/apy`;
  const res = await fetch(url, {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
    },
  });

  if (!res.ok) {
    throw new Error(`Failed to fetch BlazeStake APY: ${res.status} ${res.statusText}`);
  }

  const data = await res.json() as SolBlazeApyResponse;
  return data;
}

// Example usage
(async () => {
  try {
    const apyResp = await getSolBlazeApy();
    console.log("BlazeStake APY:", apyResp.apy, "success:", apyResp.success);
  } catch (err) {
    console.error("Error fetching BlazeStake APY:", err);
  }
})();

File: data_plane/apy_bsol.py
import requests
from typing import TypedDict, Optional

class SolBlazeApyResponse(TypedDict):
    success: bool
    apy: float  

BASE_URL = "https://stake.solblaze.org"

def get_solblaze_apy() -> SolBlazeApyResponse:
    url = f"{BASE_URL}/api/v1/apy"
    resp = requests.get(url)
    resp.raise_for_status()
    return resp.json()

if __name__ == "__main__":
    try:
        result = get_solblaze_apy()
        print("BlazeStake APY:", result["apy"], "success:", result["success"])
    except Exception as e:
        print("Error:", e)
